lib/facility.rb

```diff
@@ -1,14 +1,68 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    if services.include?('Vehicle Registration')
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Date.today
+      fees_and_plates(vehicle)
+      @registered_vehicles
+    else
+      nil
+    end
+  end
+
+  def fees_and_plates(vehicle)
+    if vehicle.antique?
+      @collected_fees += 25
+      vehicle.plate_type = :antique
+    elsif vehicle.electric_vehicle?
+      @collected_fees += 200
+      vehicle.plate_type = :ev
+    else
+      @collected_fees += 100
+      vehicle.plate_type = :regular
+    end
+  end
+
+  def administer_written_test(registrant)
+    if services.include?('Written Test')
+      if registrant.permit? && registrant.age >= 16
+        registrant.license_data[:written] = true
+      else 
+        false
+      end
+    else
+      false
+    end
```

Comment:
Nested `if`s can get buggy and hard to test every single outcome. You should actually be able to bring all the conditionals up to the first `if` for the same outcome:
```suggestion
    if services.include?('Written Test') && registrant.permit? && registrant.age >= 16
      registrant.license_data[:written] = true
    else
      false
    end
```